sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -995,32 +1126,41 @@ Exact location of the file and final name of the compiler option is TBD.
 
 ##### AOT target identification
 
-There are several user-visible places in the SDK where SYCL device target
-architectures need to be identified:
+There are several user-visible places in the SDK where SYCL target device needs to
+be identified:
 
 - `-fsycl-targets` option
 - a device configuration file entry
 - `-target` option of the `sycl-aspec-filter` tool
 - a SYCL aspect enum identifier (we expect to add a new SYCL aspect for each
-  device target architecture)
-
-In all such places architecture naming should be the same. In some cases aliases
-are allowed. Below is a list of target architectures supported by DPC++:
-
-| target/alias(es) | description |
-|-|-|
-| intel_gpu    | Generic Intel graphics architecture |
-| intel_gpu_tgl, intel_gpu_12_0 | Intel Tiger Lake (11th generation Core) integrated graphics architecture |
-| ptx64  | Generic 64-bit PTX target architecture |
-| spir64 | Generic 64-bit SPIR-V target |
-| x86_64 | Generic 64-bit x86 architecture |
-
-TODO: Provide full list of AOT targets supported by the identification
-mechanism.
+  device target)
+
+In all such places the naming scheme should be the same. In some cases aliases
+are allowed. Below is a table describing targets supported by DPC++.
+A target can fall into two logical categories:
+- An Instruction Set Architecture (ISA) version or ID, identifies just the
+  instruction set supported by the target.
+- A platform ID, identifies a concrete product configuration. E.g. there can be
+  two platforms with different cache size, but the same ISA. So platform ID is
+  provides more specific info about the target.
+
+| target/alias(es)           | ISA Id           | description                                                    |
+|----------------------------|------------------|----------------------------------------------------------------|
+| intel_gpu                  |                  | Generic Intel GPU                                              |
+| intel_gpu_skl              | intel_gpu_9_0_0  | Intel 6th generation Core (Skylake) integrated GPU             |
+| intel_gpu_kbl              | intel_gpu_9_1_0  | Intel 7th generation Core (Kaby Lake) integrated GPU           |
+| intel_gpu_cfl              | intel_gpu_9_2_0  | Intel 8th/9th generation Core (Coffee Lake) integrated GPU     |
+| intel_gpu_icl              | intel_gpu_11_0_0 | Intel 10th generation Core (Ice Lake) integrated GPU           |
+| intel_gpu_tgl              | intel_gpu_12_0_0 | Intel 11th generation Core (Tiger Lake) integrated GPU         |
+| ptx64                      |                  | Generic 64-bit PTX target architecture                         |
+| spir64                     |                  | Generic 64-bit SPIR-V target                                   |
```

Comment:
You are right about spir64 - it is different. We want to make it possible to specify it on the command line in -fsycl-targets, but don't want in the config file. I can add this comment in the 'description' column saying:
```
Generic 64-bit SPIR-V target. This is a predefined target known to to the compiler, which must not be defined in the device configuration file. specifying it in `-fsycl-targets=spir64,...` makes the compiler add the device-neutral SPIR-V binary resulted from linking device code into the resulting fat binary. It can be used as a fallback by the SYCL runtime when needed.
```

Reply:
I think `intel_gpu`, `spir64`, and `ptx64` are all similar because they are allowed only as a parameter to `-fsycl-targets` (i.e. they don't have entries in the config file).  Maybe it's better to document them in  a separate table / list, rather than via a new column in this table.

> ptx64 is a real AOT target producing a binary runnable on any (from the set supported by OneAPI) CUDA device.

Isn't this analogous to `spir64`, though?  For example, we don't know the set of aspects that are defined for the `ptx64` target.  Therefore, it seems like `ptx64` wouldn't be an entry in the config file.

Reply:
> we don't know the set of aspects

I think we should know and define this set actually.

Reply:
> I think we should know and define this set actually.

This will only be possible if all devices that use PTX have exactly the same set of aspects.  I presume this won't be the case, will it?

I thought PTX was an IL similar to SPIR-V?  When we compile to SPIR-V, we don't know the aspects because we don't know which device the IL will be compiled for.  Isn't PTX the same?